paimon-core/src/main/java/org/apache/paimon/tag/Tag.java

```diff
@@ -0,0 +1,203 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.paimon.tag;
+
+import org.apache.paimon.Snapshot;
+import org.apache.paimon.fs.FileIO;
+import org.apache.paimon.fs.Path;
+import org.apache.paimon.utils.JsonSerdeUtil;
+
+import org.apache.paimon.shade.jackson2.com.fasterxml.jackson.annotation.JsonCreator;
+import org.apache.paimon.shade.jackson2.com.fasterxml.jackson.annotation.JsonGetter;
+import org.apache.paimon.shade.jackson2.com.fasterxml.jackson.annotation.JsonProperty;
+
+import javax.annotation.Nullable;
+
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.time.Duration;
+import java.time.LocalDateTime;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Optional;
+
+import static org.apache.paimon.CoreOptions.DEFAULT_TAG_TIME_RETAINED;
+
+/** Snapshot with tagCreateTime and tagTimeRetained. */
+public class Tag extends Snapshot {
+
+    private static final String FIELD_TAG_CREATE_TIME = "tagCreateTime";
+    private static final String FIELD_TAG_TIME_RETAINED = "tagTimeRetained";
+    private static final LocalDateTime DEFAULT_TAG_CREATE_TIME =
+            LocalDateTime.of(1970, 1, 1, 0, 0, 0, 0);
+
+    @JsonProperty(FIELD_TAG_CREATE_TIME)
+    @Nullable
+    private final LocalDateTime tagCreateTime;
```

Comment:
This field will cause old readers to be unable to read the new format, which can be a bit troublesome. Can we just use file creation time?

Reply:
I have maked auto-create-tag use Snapshot json, not new Tag json.